framework/files/system/oem/01_ros-rootfs.yaml

```diff
@@ -8,6 +8,17 @@ stages:
           echo rancher-${RANDOM} > /usr/local/etc/hostname
         fi
         ln -sf /usr/local/etc/hostname /etc/hostname
+    - if: '[ ! -f "/run/cos/recovery_mode" ]'
+      name: "Persist /etc/machine-id"
```

Comment:
Like above we could think up of setting persistent files in some sort of generic approach, even I do not see it as obvious as the boot mode sentinel file.

Persisting files/folder can easily require some additional custom configs or tweaks.